elements/lisk-transactions/src/create_sendable.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { validateAddress, validatePublicKey } from '@liskhq/lisk-validator';
+
+import { TransactionJSON } from './transaction_types';
+
+import {
+	getAddressFromPublicKey,
+	getPrivateAndPublicKeyFromPassphrase,
+} from '@liskhq/lisk-cryptography';
+
+export const validateObligatoryInputs = (
+	type: number,
+	passphrase: string,
+	receipientId?: string | null,
+	recipientPublicKey?: string,
+): void => {
+	if (!(receipientId || recipientPublicKey)) {
+		throw new Error(
+			'Either recipientId or recipientPublicKey must be provided.',
+		);
+	}
+
+	if (typeof receipientId !== 'undefined' && receipientId !== null) {
+		validateAddress(receipientId);
+	}
+
+	if (typeof recipientPublicKey !== 'undefined') {
+		validatePublicKey(recipientPublicKey);
+	}
+
+	if (
+		receipientId &&
+		recipientPublicKey &&
+		receipientId !== getAddressFromPublicKey(recipientPublicKey)
+	) {
+		throw new Error('recipientId does not match recipientPublicKey.');
+	}
+
+	if (!passphrase) {
+		throw new Error(
+			'Cannot sign a transaction without a passphrase. Specify your passphrase as in the input object (and optional second passphrase)',
+		);
+	}
+
+	if (typeof type !== 'number') {
+		throw new Error('type must be provided.');
+	}
+};
+
+const skipUndefined = (
+	transaction: Partial<TransactionJSON>,
+): Partial<TransactionJSON> =>
+	// tslint:disable-next-line no-inferred-empty-object-type
+	Object.entries(transaction).reduce(
+		(transactionWithValues: object, [property, value]) => {
+			if (value !== undefined) {
+				Object.assign(transactionWithValues, { [property]: value });
+			}
+
+			return transactionWithValues;
+		},
+		{},
+	);
+
+export const createSendable = (
+	// tslint:disable-next-line
+	Transaction: any,
+	inputs: Partial<TransactionJSON>,
+	passphrase: string,
+	secondPasshprase: string,
+): string => {
+	const type = Transaction.TYPE;
+	const {
+		amount,
+		asset,
+		fee,
+		recipientId,
+		recipientPublicKey,
+		senderPublicKey,
+		timestamp,
+	} = inputs;
+	validateObligatoryInputs(type, passphrase, recipientId, recipientPublicKey);
+	
+	const transaction = new Transaction({
+		asset: asset || {},
+		amount,
+		fee,
+		recipientId: 
+			recipientId || getAddressFromPublicKey(recipientPublicKey)
+		senderPublicKey:
+			senderPublicKey || getPrivateAndPublicKeyFromPassphrase(passphrase),
+		type,
+		timestamp: timestamp || 0,
+	});
+
+	transaction.sign(passphrase, secondPasshprase);
+
+	return JSON.stringify(skipUndefined(transaction.toJSON()));
```

Comment:
`JSON.stringify` removes keys with `undefined` values.
No need to use `skipUndefined ` here.